src/exchange/passthrough.go

```diff
@@ -323,34 +387,60 @@ beginOrderLoop:
 		case <-time.After(checkOrderWait):
 			status, err = p.checkOrder(orderID)
 			if err != nil {
+				log.WithError(err).Error("checkOrder failed")
 				return di, err
 			}
 
+			log.WithFields(logrus.Fields{
+				"orderID":     orderID,
+				"orderStatus": status,
+			}).Info("Got order status")
+
 			switch status {
 			case exchange.Completed:
 			default:
 				continue beginOrderLoop
 			}
 		}
 
+		// Calculate the number of SKY bought, measured in droplets
+		skyBought := ask.Volume.Mul(decimal.New(droplet.Multiplier, 0)).IntPart()
+		if skyBought < 0 {
+			err := errors.New("Calculated SKY bought is negative")
+			log.WithError(err).Error("CRITICAL ERROR")
+			return di, err
+		}
+
+		// Calculate number of satoshis spent
+		// TODO -- review the math to make sure the truncation caused by IntPart()
+		// does not lead us to have a tiny unspendable amount of BTC remaining
+		// that prevents us from exiting the loop.
+		// A skycoin droplet has a minimum price measured in BTC -- this may be unavoidable
+		// and the loop check may have to account for this possibility.
+		// Also, we only purchase skycoin to a precision that we are able to send,
+		// defined by visor.MaxDropletPrecision, which is 3 at the time of this writing.
+		// So the smallest purchasable unit is 0.001 SKY.
+		// If the BTC/SKY price is 0.0012, then the minimum amount of BTC to spend
+		// is 0.0012 * 0.001 = 0.0000012 BTC, or 120 satoshis.
+		spent := askCost(ask).Mul(decimal.New(SatoshisPerBTC, 0)).IntPart()
```

Comment:
instead of having satoshi per btc as 8 maybe we should have it according to what maximum an exchange supports